presto-main/src/main/java/com/facebook/presto/sql/planner/LogicalPlanner.java

```diff
@@ -130,29 +130,35 @@ public Plan plan(Analysis analysis, Stage stage)
 
     public PlanNode planStatement(Analysis analysis, Statement statement)
     {
-        if (statement instanceof CreateTableAsSelect) {
+        if (statement instanceof CreateTableAsSelect && analysis.isCreateTableAsSelectNoOp()) {
             checkState(analysis.getCreateTableDestination().isPresent(), "Table destination is missing");
+            List<Expression> emptyRow = ImmutableList.of();
+            PlanNode source = new ValuesNode(idAllocator.getNextId(), ImmutableList.of(), ImmutableList.of(emptyRow));
+            return new OutputNode(idAllocator.getNextId(), source, ImmutableList.of(), ImmutableList.of());
+        }
+        return createOutputPlan(planStatementWithoutOutput(analysis, statement), analysis);
+    }
+
+    private RelationPlan planStatementWithoutOutput(Analysis analysis, Statement statement)
+    {
+        if (statement instanceof CreateTableAsSelect) {
             if (analysis.isCreateTableAsSelectNoOp()) {
-                List<Expression> emptyRow = ImmutableList.of();
-                PlanNode source = new ValuesNode(idAllocator.getNextId(), ImmutableList.of(), ImmutableList.of(emptyRow));
-                return new OutputNode(idAllocator.getNextId(), source, ImmutableList.of(), ImmutableList.of());
-            }
-            else {
-                return createOutputPlan(createTableCreationPlan(analysis, ((CreateTableAsSelect) statement).getQuery()), analysis);
+                throw new PrestoException(NOT_SUPPORTED, "CTAS with NoOp statement is not supported in this context " + statement.getClass().getSimpleName());
```

Comment:
This method return type is different: `RelationPlan` in ..withoutOutput() method vs `PlanNode` in planStatement(). So, if I want to return just a ValuesNode then I need to create a `RelationPlan` around it which seems like an unnecessary and messy change. Also, I'm not sure what `scope` should be in this case.